spark_pipeline_framework/transformers/fhir_sender/v1/fhir_sender.py

```diff
@@ -364,6 +383,32 @@ def _transform(self, df: DataFrame) -> DataFrame:
                 desired_partitions: int = num_partitions or math.ceil(
                     row_count / batch_size
                 )
+                if sort and sort.get("partition_by_column_name"):
+                    json_df.withColumn(
+                        sort["partition_by_column_name"],
+                        get_json_object(
+                            col("value"), f"$.{sort['partition_by_column_name']}"
+                        ),
+                    ).repartition(sort["partition_by_column_name"])
+                    if sort.get("column_for_sorting"):
+                        json_df.sortWithinPartitions(sort["column_for_sorting"])
+                    json_df = json_df.drop(sort["partition_by_column_name"])
+                    desired_partitions = json_df.rdd.getNumPartitions()
```

Comment:
this is not "desired" partitions. If we have desired_partitions value, we should pass it to repartition.